core/src/test/java/io/grpc/ClientInterceptorsTest.java

```diff
@@ -351,6 +352,33 @@ protected void checkedStart(ClientCall.Listener<RespT> responseListener, Metadat
     ArgumentCaptor<Status> captor = ArgumentCaptor.forClass(Status.class);
     verify(listener).onClose(captor.capture(), any(Metadata.class));
     assertSame(error, captor.getValue().getCause());
+
+    // Make sure nothing bad happens after the exception.
+    ClientCall<?, ?> noop = ((CheckedForwardingClientCall<?, ?>)interceptedCall).delegate();
```

Comment:
That is what the `interceptedCall.*` lines above were doing. From what I can tell, just `cancel()` and `isReady()` need to be called, and those can be done with `interceptedCall` quite naturally. The `verifyNotMoreInteractions(call)` is quite important to actually verifying the correct behavior.

If you want to test NOOP_CALL more, I'd suggest just making it package-private and calling methods on it directly, possibly just adding `assertSame(NOOP_CALL, noop)` here. But even if you did that I'd hope that `cancel()` and `isReady()` were still called on `interceptedCall`.